Finmer.Editor/FurballContentReaderText.cs

```diff
@@ -28,12 +29,26 @@ public sealed class FurballContentReaderText : IFurballContentReader
         private readonly DirectoryInfo m_SearchPath;
         private readonly Stack<JToken> m_TokenStack = new Stack<JToken>();
         private readonly Stack<JToken> m_ArrayStack = new Stack<JToken>();
+        private readonly int m_Version;
         private JToken m_CurrentArrayElement;
 
+        public FurballContentReaderText(JObject root, DirectoryInfo searchPath, int version)
+        {
+            m_SearchPath = searchPath;
+            m_TokenStack.Push(root);
+            m_Version = version;
+        }
+
         public FurballContentReaderText(JObject root, DirectoryInfo searchPath)
         {
             m_SearchPath = searchPath;
             m_TokenStack.Push(root);
+            m_Version = ReadInt32Property("FormatVersion");
```

Comment:
This only works if the JSON document being read is the project metadata file. So this ctor has a subtle precondition.

Perhaps instead this ctor could leave the version uninitialized (e.g. set to zero), and we add a setter. The getter can throw if the version field is set to zero.

Alternatively, perhaps a prettier solution, is to add a static `FromProjectMetadata` factory method, so that the precondition is explicitly documented.